src/Xamarin.Android.Build.Tasks/Tasks/GetAdditionalResourcesFromAssemblies.cs

```diff
@@ -172,7 +173,7 @@ bool ExtractArchive (string url, string file, string contentDir)
 					}
 				}
 				catch (Exception e) {
-					LogCodedError ("XA5209", "Unzipping failed. Please download {0} and extract it to the {1} directory.", url, contentDir);
+					LogCodedError ("XA5209", "Unzipping failed while attempting to extract {0} into {1}{2}", file, contentDir, Path.DirectorySeparatorChar);
```

Comment:
After considering this a little more, I think depending on https://github.com/xamarin/xamarin-android/pull/150 to provide the "not a ZIP" message is indeed probably the simplest approach.  That might also mean that the 2 separate "XA5209" lines can now be condensed into just 1 line with the "Reason:" at the end (there is some precedent for that).

I'll aim to add a "second draft" onto this pull request by Monday that incorporates the various notes so far.  Thanks!